### ISSUE TITLE
useless-super-delegation when an override changes the return type annotation

### ISSUE THREAD

**Issue by Infernio** [PyCQA/pylint#5822]
### Current problem

Similar to #1923, but for the return type annotation.
It is sometimes useful to override a method and change the return type annotation - whether to help a static type checker out or to help the IDE make better autocomplete suggestions.

Here's an example - it's a bit contrived, I actually ran into this when overriding to add typing to a method from a third-party library (ANTLR).

```python
"""Quick test to demonstrate useless-super-declaration raised for overrides
that change return type annotations."""
import random
from dataclasses import dataclass
from typing import Any

# pylint: disable=too-few-public-methods
class _AGenerator:
    """Abstract base class for generators."""
    _valid_values: list[Any]

    def get_random(self) -> Any:
        """Generates a random value from the valid values that this generator
        can return."""
        return random.choice(self._valid_values)

@dataclass
class StrGenerator(_AGenerator):
    """Generator that accepts and returns string values."""
    _valid_values: list[str]

    # override to add typing
    def get_random(self) -> str:
        return super().get_random()

s = StrGenerator(['foo', 'bar', 'qux'])
print(s.get_random())
print(s.get_random())
print(s.get_random())
```

Running pylint with default config on it:
```
$ python -m pylint test.py 
************* Module test
test.py:24:4: W0235: Useless super delegation in method 'get_random' (useless-super-delegation)

------------------------------------------------------------------
Your code has been rated at 9.33/10 (previous run: 9.33/10, +0.00)
```

### Desired solution

pylint should not emit a warning in this case as the override has legitimate uses.

**Comment by JonnoFTW** [comment]
I experience the same error, except that the superclass is an ABC. I would consider this a bug seeing as this behaviour contradicts what is in the changelog for 2.0
